File: 程式設計：資料結構答案檔/C#/101.cs
using System;
using System.Collections.Generic;
using System.Linq;

class Program
{
    static void CalculateSumAvg(List<List<string>> inputs)
    {
        foreach (var studentScores in inputs)
        {
            var scores = studentScores.Skip(1).Select(int.Parse).ToList();
            var sumScore = scores.Sum();
            var avgScore = (double)sumScore / scores.Count;
            studentScores.Add(sumScore.ToString());
            studentScores.Add($"{avgScore:F2}");
        }
    }

    static void Main(string[] args)
    {
        var inputs = new List<List<string>>();
        string inputStr;
        
        while ((inputStr = Console.ReadLine()) != "END")
        {
            var inputScores = inputStr.Split().ToList();
            inputs.Add(inputScores);
        }

        CalculateSumAvg(inputs);

        inputs.Sort((x, y) => -float.Parse(x[^2]).CompareTo(float.Parse(y[^2])));

        foreach (var studentScores in inputs)
        {
            Console.WriteLine(string.Join(" ", studentScores));
        }
    }
}

File: 程式設計：資料結構答案檔/Python/101.py
def caculate_sum_avg(inputs):
    for student_scores in inputs:
        scores = list(map(int, student_scores[1:]))
        sum_score = sum(scores)
        avg_score = sum_score / len(scores)
        student_scores.append(sum_score)
        student_scores.append(f'{avg_score:.2f}')

inputs = []
while True:
    input_str = input()
    if input_str == 'END':
        break
    input_scores = input_str.split()
    inputs.append(input_scores)

caculate_sum_avg(inputs)

inputs.sort(key=lambda x: -float(x[-2]))

for student_scores in inputs:
    print(*student_scores)